src/Pages/Account/Account.js

```diff
@@ -0,0 +1,421 @@
+      }),
+    );
+  };
+
+  // 버튼 클릭시 실행되는 메소드
+  handleBtn = () => {
+    // 원래 조건 넣고 setState하는 자리
+    const {
+      firstname,
+      firstnamehasValue,
+      lastname,
+      lastnamehasValue,
+      email,
+      emailCondition,
+      emailhasValue,
+      password,
+      pwLengthCondition,
+      passwordhasValue,
+      checkPw,
+      checkpwCondition,
+      checkPwhasValue,
+      adagreement,
+      serviceagreement,
+      serviceagreementhasValue,
+    } = this.state;
+
+    const emailcheck = /^[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*@[0-9a-zA-Z]([-_.]?[0-9a-zA-Z])*.[a-zA-Z]{2,3}$/i.test(
+      email,
+    );
+    const passwordcheck = /^(?=.*?[A-Z])(?=.*?[a-z])(?=.*?[0-9])(?=.*?[#?!@$%^&*-]).{8,}$/.test(
+      password,
+    );
+
+    const checkpwcheck = password === checkPw;
+
+    const allAcceptedAccountCondition =
+      firstname &&
+      lastname &&
+      emailcheck &&
+      passwordcheck &&
+      checkpwcheck &&
+      serviceagreement;
+
+    if (!allAcceptedAccountCondition) {
+      alert("실패");
+
+      if (!firstname) {
+        this.setState({
+          firstnamehasValue: false,
+        });
+      } else if (firstname) {
+        this.setState({
+          firstnamehasValue: true,
+        });
+      }
+
+      if (!lastname) {
+        this.setState({
+          lastnamehasValue: false,
+        });
+      } else if (lastname) {
+        this.setState({
+          lastnamehasValue: true,
+        });
+      }
+
+      if (!email) {
+        this.setState({
+          emailhasValue: false,
+        });
+      } else if (email) {
+        this.setState({
+          emailhasValue: true,
+        });
+      }
+
+      if (!password) {
+        this.setState({
+          passwordhasValue: false,
+        });
+      } else if (password) {
+        this.setState({
+          passwordhasValue: true,
+        });
+      }
+
+      if (!checkPw) {
+        this.setState({
+          checkPwhasValue: false,
+        });
+      } else if (checkPw) {
+        this.setState({
+          checkPwhasValue: true,
+        });
+      }
+
+      if (!serviceagreement) {
+        this.setState({
+          serviceagreementhasValue: false,
+        });
+      } else if (serviceagreement) {
+        this.setState({
+          serviceagreementhasValue: true,
+        });
+      }
+    }
+
+    fetch(SIGNUP_API, {
+      method: "POST",
+      body: JSON.stringify({
+        firstname: firstname,
+        lastname: lastname,
+        email: email,
+        password: password, // 로그인은 pw로 함.
+        checkpassword: checkPw,
+        adagreement: adagreement,
+      }),
+    })
+      .then((response) => response.json())
+      .then((result) => {
+        alert({ result });
+
+        //
+        if (result.message === "SUCCESS") {
+          //localStorage.setItem("token", result.Authorization);
+          this.props.history.push("/"); //2주차에 마이페이지
+          return;
+        }
+
+        alert(실패);
+      });
```

Comment:
```suggestion
    () => this.fetchSignUp();
```
- 기능 단위로 함수를 분리해주시기 바랍니다!
- 원래는 validation 함수에서 emailcheck, pwcheck 전부 해주고, 버튼 클릭시에 fetch 함수 실행하는게 옳겠지만 우선은 이 정도로 마무리해도 좋을 것 같습니다 :)

Reply:
수정했습니다. 

다만 2번째 줄에 말씀하신 것을 제가 제대로 이해하고 실행했는지 모르겠습니다.
validation 함수(`checkValidity` 메소드)에서 emailcheck, pwcheck 전부 해주고 왔는데 "버튼 클릭시에 fetch 함수 실행"한다는 부분과는 다르게 제가 버튼 클릭 후 별도로 fetch 함수를 분리(`this.fetchSignUp()`)시켜주어서 제가 더 리팩토링을 하는 것이 좋을지 궁금합니다. 

감사합니다 :)